signer/src/request_decider.rs

```diff
@@ -236,8 +254,8 @@ where
         // TODO: Do we want to do this on the sender address or the
         // recipient address?
         let is_accepted = self
-            .can_accept(&withdrawal_request.sender_address.to_string())
-            .await;
+            .can_accept_withdrawal_request(&withdrawal_request)
+            .await?;
```

Comment:
nit: I like that the` can_accept_*` functions now consistently return a Result. However, it seems this particular call can't fail. The `?` threw me off, making me wonder where it might fail. Maybe consider using expect to make it clear this is infallible? On the cons side, If the function ever changes to return an error, we'd need to update the expect. Just a thought, happy to leave it as-is if you prefer!

Reply:
Yeah the infallible scenario is just because this is anemic compared to its deposit equivalent, so I fully expect this to be built out as well when we get there.